Python/pythonrun.c

```diff
@@ -431,6 +431,9 @@ PyRun_SimpleFileExFlags(FILE *fp, const char *filename, int closeit,
     }
     flush_io();
     if (v == NULL) {
+        if (PyErr_ExceptionMatches(PyExc_SystemExit)) {
+            Py_DECREF(m);
```

Comment:
Why make this a special case? We could just always `Py_DECREF(m)` early.

Reply:
No. `Py_DECREF(m)` would then be called one too many times.